sale_general_discount/views/sale_order_view.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<!-- Copyright 2018 Tecnativa - Sergio Teruel
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record id="sale_order_general_discount_form_view" model="ir.ui.view">
+        <field name="name">sale.order.general.discount.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='amount_tax']" position="after">
+                <field name="discount"/>
+            </xpath>
+            <xpath expr="//field[@name='order_line']" position="attributes">
+                <attribute name="context">{'default_discount': discount}</attribute>
```

Comment:
I prefer the method I showed you through `fields_view_get` to be resilient to other modules modifying the same context. I'm not sure if we can use the new XML attribute `add` for adding the context here. See an example in:

https://github.com/odoo/odoo/blob/7526b64e8a1cf62c2a22e75c0c051a4abad67ec6/addons/website_blog/views/website_blog_templates.xml#L207

Reply:
I tried it, but only join as string, you can not update a dictionary

Reply:
Thanks for telling. I will check if it's possible to use it with little changes and propose them to Odoo.